cli/src/commands/fix.rs

```diff
@@ -328,3 +390,126 @@ fn run_tool(
+                let table = val.into_table()?;
+                let name = table
+                    .get("name")
+                    .ok_or(config::ConfigError::Message(format!(
+                        "`fix.tools` entry at index {} is missing required field `name`.",
+                        index
+                    )))?
+                    .to_string();
+                if let Some(first_index) = tool_name_first_seen_index.insert(name.clone(), index) {
+                    return Err(config::ConfigError::Message(format!(
+                        "`fix.tools` entries at indices {} and {} have the same `name`: {}",
+                        first_index, index, name
+                    ))
+                    .into());
+                }
+                let patterns: Vec<String> = table
+                    .get("patterns")
+                    .ok_or(config::ConfigError::Message(format!(
+                        "`fix.tools` entry `{}` at index {} is missing required field `patterns`.",
+                        name, index
+                    )))?
+                    .clone()
+                    .into_array()?
+                    .into_iter()
+                    .map(|value| -> Result<String, ConfigError> { value.into_string() })
+                    .try_collect()?;
+                if patterns.is_empty() {
+                    Err(config::ConfigError::Message(format!(
+                        "`fix.tools` entry `{}` at index {} must have at least one pattern.",
+                        name, index
+                    ))
+                    .into())
+                } else {
+                    Ok(ToolConfig {
+                        command: CommandNameAndArgs::Vec(
+                            NonEmptyCommandArgsVec::try_from(
+                                table
+                                    .get("command")
+                                    .ok_or(config::ConfigError::Message(format!(
+                                        "`fix.tools` entry `{}` at index {} is missing required \
+                                         field `command`.",
+                                        name, index
+                                    )))?
+                                    .clone()
+                                    .into_array()?
+                                    .iter()
+                                    .map(|value| value.to_string())
+                                    .collect_vec(),
+                            )
+                            .map_err(|e| {
+                                config::ConfigError::Message(format!(
+                                    "`fix.tools` entry `{}` at index {}: {}",
+                                    name, index, e
+                                ))
+                            })?,
+                        ),
+                        matcher: workspace_command
+                            .parse_union_filesets(&patterns)?
```

Comment:
nit: this will resolve patterns relative to cwd, not to workspace root. I'm not sure if that's correct or wrong.